src/NHibernate.Test/TestExtensions.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace NHibernate.Test
+{
+	//Utility extension methods for use in unit tests. 
+	public static class TestExtensions
+	{
+		public static T SetPropertyUsingReflection<T, TProp>(this T instance, Expression<Func<T, TProp>> property, TProp value)
+		{
+			var method = property.Body as MemberExpression;
+			var propertyInfo = typeof(T).GetProperty(method.Member.Name);
+			if (propertyInfo!= null && propertyInfo.CanWrite)
+			{
+				propertyInfo.SetValue(instance, value);
+			}
+			else
+			{
+				//camel cased field
+				var name = method.Member.Name.Substring(0, 1).ToLowerInvariant() + method.Member.Name.Substring(1);
```

Comment:
This was a method I just copied straight from my own at-work code, where we usually use it to set properties on NH mapped domain classes, where the setters are hidden. It uses the current form to provide chaining, so that settings many properties doesn't become a burden. The lambda is just refactoring friendly (we don't use the field version, it was added here)

I don't mind having a simpler implementation. 

> At the very least if we keep this, it should throw when not finding the field.

Absolutely. That was an omission.

Reply:
So maybe adjust it by changing
```c#
if (field != null)
{
	field.SetValue(instance, value);
}
```

To:
```c#
if (field == null)
    var field = typeof(T).GetField("_" + name, BindingFlags.NonPublic | BindingFlags.Instance);
if (field == null)
    throw new InvalidOperationException($"Unable to find a setter or a backing field for property {method.Member.Name}");

field.SetValue(instance, value);
```